pytensor/tensor/subtensor.py

```diff
@@ -197,18 +269,47 @@ def get_canonical_form_slice(
     in which 0 <= start <= stop <= length and step > 0, and a flag which says
     if the resulting set of numbers needs to be reversed or not.
 
+    Given a scalar index `idx` that may or not be negative, convert it to
+    a certainly positive form `idx if idx >= 0 else length + idx`.
+
+    Returns
+    -------
+    slc
+        Canonical form slice or scalar variable.
+    direction
+        Direction to iterate the resulting elements in. (-1 or 1). May be symbolic.
     """
     from pytensor.tensor import ge, lt, sign, switch
 
-    if not isinstance(theslice, slice):
-        try:
-            value = as_index_literal(theslice)
-        except NotScalarConstantError:
-            value = theslice
-
-        value = switch(lt(value, 0), (value + length), value)
+    # Convert the two symbolic slice types into a native slice
+    if isinstance(theslice, SliceConstant):
```

Comment:
I still think this function is not supposed to handle non-python slices (be it constant or not).

That allows us to simplify our logic and type hints quite a lot?

Reply:
As long as symbolic slices exist, I think this function should support them. How else should they be handled?

For now, I'd just like to get the typing fixed.

Reply:
This function does not seem to be intended for SliceTypes. I wrote elsewhere but there are two ways of representing slices in PyTensor: basic index operations don't have explicit slice inputs, they just take as inputs the start/stop/step variables that define a slice.

Advanced indexing operations instead habe explicit slice variables os SliceVariable type as inputs. This function was seemingly not meant to work with those types and unless that's why you went into this refactor there's no reason to extend it's behavior to handle types that are ony used by advanced indexing operations.

Does that make sense?